packages/file-asset-apis/src/s3/s3-helpers.ts

```diff
@@ -9,9 +9,45 @@ import {
     CompleteMultipartUploadCommand, AbortMultipartUploadCommand,
 } from '@aws-sdk/client-s3';
 
-import { TSError } from '@terascope/utils';
+import { TSError, pDelay, AnyObject } from '@terascope/utils';
 import { S3ClientParams, S3ClientResponse } from './client-types';
 
+type S3RetryResponse = S3ClientResponse.ListObjectsV2Output | S3ClientResponse.GetObjectOutput;
+type S3RetryParams = S3ClientParams.ListObjectsV2Request | S3ClientParams.GetObjectRequest;
+
+export async function s3RequestWithRetry<P extends S3RetryParams>(
+    client: S3Client,
+    func: (
+        client: S3Client,
+        params: P
```

Comment:
this is ok, but it would be better if this was implemented using [function overloading](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads). That way it should be smarter about it outputs and so you don't have to override the output of this everywhere you use it